### DISCUSSION TITLE
Example in apigatewaymanagementapi post-to-connection  failed

### DISCUSSION THREAD

**Discussion by johnc-c** [aws/aws-cli#6984] [Q&A] [answered]
### Describe the bug

Got the error when running the example in the awscli/examples/apigatewaymanagementapi/post-to-connection.rst
`aws apigatewaymanagementapi post-to-connection --connection-id L0SM9cOFvHcCIhw= --data "Hello from API Gateway!" --endpoint-url https://aabbccddee.execute-api.us-west-2.amazonaws.com/prod`

`Invalid base64: "Hello from API Gateway!"`

### Expected Behavior

`Could not connect to the endpoint URL:"https://aabbccddee.execute-api.us-west-2.amazonaws.com/prod/@connections/L0SM9cOFvHcCIhw%3D"`

### Current Behavior

`aws apigatewaymanagementapi post-to-connection --connection-id L0SM9cOFvHcCIhw= --data "Hello from API Gateway!" --endpoint-url https://aabbccddee.execute-api.us-west-2.amazonaws.com/prod`

`Invalid base64: "Hello from API Gateway!"`

### Reproduction Steps

Run the cmd 

`aws --version`

`aws-cli/2.1.18`

`aws apigatewaymanagementapi post-to-connection --connection-id L0SM9cOFvHcCIhw= --data "Hello from API Gateway!" --endpoint-url https://aabbccddee.execute-api.us-west-2.amazonaws.com/prod`

`Invalid base64: "Hello from API Gateway!"`

### Possible Solution

It worked after adding the "--cli-binary-format".

`aws apigatewaymanagementapi post-to-connection --connection-id L0SM9cOFvHcCIhw=  --cli-binary-format raw-in-base64-out --data "Hello from API Gateway!" --endpoint-url https://aabbccddee.execute-api.us-west-2.amazonaws.com/prod`

`Could not connect to the endpoint URL: "https://aabbccddee.execute-api.us-west-2.amazonaws.com/prod/@connections/L0SM9cOFvHcCIhw%3D`

### CLI version used

aws-cli/2.1.18 Python/3.7.3 

### Environment details (OS name and version, etc.)

Linux/5.10.102.1-microsoft-standard-WSL2 exe/x86_64.ubuntu.20 prompt/off

**Comment by tim-finnigan** [comment, marked as answer]
Hi @JohnChen-K thanks for reaching out. This behavior is referenced here in the CLI v1 to v2 migration changes documentation: https://docs.aws.amazon.com/cli/latest/userguide/cliv2-migration-changes.html#cliv2-migration-binaryparam

There have been a few other issues opened regarding this such as https://github.com/aws/aws-cli/issues/4994 and https://github.com/aws/aws-cli/issues/5053. As mentioned in the documentation linked above, you can revert to the AWS CLI version 1 behavior using the `cli_binary_format` file configuration or the `--cli-binary-format` parameter. I'm going to convert this issue to a Discussion per our guidance process.
